### PR TITLE
chore: adding proxy test cases for salesforce oauth

### PR DESCRIPTION
**Pull request by shrouti1507**
## What are the changes introduced in this PR?

We have already added test cases for Salesforce. As Salesforce Oauth uses the same code base, except network handler ( in case of invalid session id ) we are adding a test case for the refresh token scenario.
For this oauth flow of the destination no test cases were added previously.

As no test cases were added previously in this flow. So new number of test cases: 2
Vulnerabilities caught : 0

## What is the related Linear task?

Resolves INT-1493

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [x] My code follows the style guidelines of this project

- [x] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [x] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [x] Is the PR limited to 10 file changes?

- [x] Is the PR limited to one linear task?

- [x] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3170/test-report.html

File: test/integrations/destinations/salesforce_oauth/dataDelivery/oauth.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  },
+  destInfo: { authKey: 'dummyDestinationId' },
+  dontBatch: false,
+};
+
+export const reqMetadataArrayWithSecret = [proxyMetdataWithSecret];
+
+export const testScenariosForV1API: ProxyV1TestData[] = [
+  {
+    id: 'salesforce_v1_scenario_8',
```
Comment (yashasvibajpai) [inline test/integrations/destinations/salesforce_oauth/dataDelivery/oauth.ts:60]: Can you fix this to
```suggestion
    id: 'salesforce_v1_scenario_1',
```

**Comment by yashasvibajpai** [comment]
@shrouti1507 seeing lint related test failing, can you check it?

File: test/integrations/destinations/salesforce_oauth/dataDelivery/oauth.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  dontBatch: false,
+};
+
+export const reqMetadataArrayWithSecret = [proxyMetdataWithSecret];
+
+export const testScenariosForV1API: ProxyV1TestData[] = [
+  {
+    id: 'salesforce_v1_scenario_1',
+    name: 'salesforce_oauth',
+    description: '[Proxy v1 API] :: Test with expired access token scenario',
```
Comment (utsabc) [inline test/integrations/destinations/salesforce_oauth/dataDelivery/oauth.ts:62]: Add a positive and invalid access case

Comment (shrouti1507) [reply test/integrations/destinations/salesforce_oauth/dataDelivery/oauth.ts:89]: expired or invalid session id response is same

**Review by utsabc — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

**Pull request merged**